system/jlib/jhash.ipp

```diff
@@ -173,6 +172,27 @@ class MappingStringTo : extends Atom
      VALUE_T    val;
 };
 
+template <class VALUE_T,class VALINIT>
+class MappingConstStringTo : public CInterfaceOf<IAtom>
+{
+public:
+    MappingConstStringTo(const char *k, VALINIT a) : key(k), val(a), hash(0)
+    { };
+
+    inline VALUE_T &            getValue()      { return val; };
+
+//interface:IMapping
+    virtual const char *        queryStr() const { return key; }
+    virtual const void *        getKey() const { return key; }
+    virtual unsigned            getHash() const { return hash; }
+    virtual void                setHash(unsigned hval) { hash = hval; }
+
+  protected:
+    VALUE_T val;
```

Comment:
trivial: may pack smaller if this was the last field (since the hash will combine with the link count), leaving the key correctly aligned.